### PR TITLE
fix: brightspace connector 500 error and automatically close dialog

### PR DESCRIPTION
**Pull request by edalex-yinzi**
##### Checklist

- [x] the [contributor license agreement][] is signed
- [x] commit message follows [commit guidelines][]
- [ ] tests are included
- [x] screenshots are included showing significant UI changes
- [ ] documentation is changed or added

##### Description of change
Force return old UI to resolve the 500 error.
Provide a custom close function to close the Dialog in the new UI.

This PR fixes two problems.

The first one is that the LMS auth dialog in the new UI will trigger a 500 error. This 500 error happens after the finishedAuth event is triggered. Since the cause of this error is difficult to trace, we force the new UI to be closed in finishedAuth Event.

The second problem was that after fixing the 500 error, the LMS auth dialog didn't close properly in the new UI. In fact, it didn't close properly even in the old UI (it disappeared because the whole brightspace connector page was redirected). The reason is the returned script `$.fancybox.close` is in the iframe, and it does not have a `self.parent` prefix or context, so this script won't close the dialog outside the iframe. And the solution was to add a correct close JS statement to the returned script.

https://user-images.githubusercontent.com/92769668/160061668-39e86656-79f4-4400-9372-0ff4e15cf1ac.mp4

~~The only issue is now the new UI will send a redundant request at the end.~~ fixed
image

[contributor license agreement]: https://www.apereo.org/node/676
[commit guidelines]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by PenghaiZhang** [comment]
As discussed, this approach may result in the warning of redundant request.

After hours of experimenting, I finally got a better solution. Let's talk on Monday.

**Comment by edalex-yinzi** [comment]
Now the redundant request issue has been fixed.

**Comment by edalex-yinzi** [comment]
Comment updated.

File: Source/Plugins/Core/com.equella.core/src/com/tle/web/connectors/dialog/LMSAuthDialog.java [hunk 1]
```diff
@@ -108,10 +119,41 @@ public void treeFinished(String id, SectionTree tree) {
...
+  public void setIsUseMockCloseDialog(boolean value) {
+    this.isUseMockCloseDialog = value;
+  }
+
+  /**
+   * Js statement for closing the dialog by clicking the close button. Because the script could be
+   * existing inside the iframe so the close function should have window.parent.ducoment prefix.
+   * Otherwise, it won't be able to close the auth dialog where outside the iframe.
+   */
+  private JQueryStatement mockCloseDialog(SectionInfo info) {
```
Comment (edalex-ian) [inline Source/Plugins/Core/com.equella.core/src/com/tle/web/connectors/dialog/LMSAuthDialog.java:135]: Not sure if this is really a 'mock', as it is actually closing the dialog - albeit kind of manually (seeing it's is calling the `click()` function). The use of mock kind of implies something not really doing something - just kind of pretending to do it.

Or am I not understanding what's going on here?

**Review by edalex-ian — COMMENTED** [review]
Some comments need tweaking, plus the naming is slightly confusing me. :thinking:

File: Source/Plugins/Core/com.equella.core/src/com/tle/web/connectors/dialog/LMSAuthDialog.java (hunk 1 above)
Comment (PenghaiZhang) [reply Source/Plugins/Core/com.equella.core/src/com/tle/web/connectors/dialog/LMSAuthDialog.java:135]: No, it's not a mock.

Comment (edalex-yinzi) [reply Source/Plugins/Core/com.equella.core/src/com/tle/web/connectors/dialog/LMSAuthDialog.java:135]: Yes, your understanding is right. How about changing the name to manualCloseDialog? Or do you have a better suggestion?

Comment (edalex-ian) [reply Source/Plugins/Core/com.equella.core/src/com/tle/web/connectors/dialog/LMSAuthDialog.java:135]: Would it be fair to call it an 'external' (to the IFRAME) auth dialog? If so, how about:

- `closeExternalAuthDialog` and
- `isExternalAuthDialog` etc

**Review by edalex-ian — APPROVED** [review]
Good stuff @edalex-yinzi , thank you. :+1:

**Review by PenghaiZhang — APPROVED** [review]

**Pull request merged**